src/utils/EditDistance.fs

```diff
@@ -23,33 +23,42 @@ let jaro (s1: string) (s2: string) =
     let matchRadius = 
         let minLen = Math.Min(s1.Length, s2.Length)
         minLen / 2 + minLen % 2
-    
-    // An inner function which recursively finds the number  
-    // of matched characters within the radius.
-    let commonChars (chars1: string) (chars2: string) =
-        let result = ResizeArray(chars1.Length)
-        for i = 0 to chars1.Length - 1 do
-            let c = chars1.[i]
-            if existsInWin c chars2 i matchRadius then 
-                result.Add c
-        result
-    
-    // The sets of common characters and their lengths as floats 
-    let c1 = commonChars s1 s2
-    let c2 = commonChars s2 s1
-    let c1length = float c1.Count
-    let c2length = float c2.Count
-        
+
+    let rec nextChar (s1:string) (s2:string) i =
+        let c = s1.[i]
+        if i < s1.Length && not (existsInWin c s2 i matchRadius) then
```

Comment:
do we need the i < s1.Length check here? looks like this is already checked outside

Reply:
scratch that. it's recursive. check is needed